adapters/impactify/impactify.go

```diff
@@ -0,0 +1,186 @@
+}
+
+type DefaultExtBidder struct {
+	Bidder openrtb_ext.ExtImpImpactify `json:"bidder"`
+}
+
+func (a *adapter) MakeRequests(bidRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	for i := 0; i < len(bidRequest.Imp); i++ {
+		// Check if imp comes with bid floor amount defined in a foreign currency
+		if bidRequest.Imp[i].BidFloor > 0 && bidRequest.Imp[i].BidFloorCur != "" && strings.ToUpper(bidRequest.Imp[i].BidFloorCur) != "USD" {
+			// Convert to US dollars
+			convertedValue, err := reqInfo.ConvertCurrency(bidRequest.Imp[i].BidFloor, bidRequest.Imp[i].BidFloorCur, "USD")
+			if err != nil {
+				return nil, []error{err}
+			}
+			bidRequest.Imp[i].BidFloorCur = "USD"
+			bidRequest.Imp[i].BidFloor = convertedValue
+		}
+
+		// Set the CUR of bid to USD after converting all floors
+		bidRequest.Cur = []string{"USD"}
+
+		var impactifyExt ImpactifyExtBidder
+
+		var defaultExt DefaultExtBidder
+		err := json.Unmarshal(bidRequest.Imp[i].Ext, &defaultExt)
+		if err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Unable to decode the imp ext : \"%s\"", bidRequest.Imp[i].ID),
+			}}
+		}
+
+		impactifyExt.Impactify = defaultExt.Bidder
+		bidRequest.Imp[i].Ext, err = json.Marshal(impactifyExt)
+		if err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Unable to decode the imp ext : \"%s\"", bidRequest.Imp[i].ID),
+			}}
+		}
+	}
+
+	if len(bidRequest.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No valid impressions in the bid request",
+		}}
+	}
+
+	reqJSON, err := json.Marshal(bidRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	if bidRequest.Device != nil {
+		if bidRequest.Device.UA != "" {
+			headers.Add("User-Agent", bidRequest.Device.UA)
+		}
+		// Add IPv4 or IPv6 if available
+		if bidRequest.Device.IP != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IP)
+		} else if bidRequest.Device.IPv6 != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IPv6)
+		}
+	}
+	if bidRequest.Site != nil {
+		headers.Add("Referer", bidRequest.Site.Page)
+	}
+
+	// set user's cookie
+	if bidRequest.User != nil && bidRequest.User.BuyerUID != "" {
+		headers.Add("Cookie", "uids="+bidRequest.User.BuyerUID)
+	}
+
+	adapterRequests = append(adapterRequests, &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+	})
+
+	return adapterRequests, nil
+}
+
+func (a *adapter) MakeBids(
+	internalRequest *openrtb2.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
```

Comment:
Please add a json test case for the 204 response code.

Reply:
Done.